src/Extensions/NexusMods.Extensions.Hashing/AbsolutePathExtensions.cs

```diff
@@ -30,4 +31,45 @@ public static class AbsolutePathExtensions
             });
     }
 
+    /// <summary>
+    /// Calculates the xxHash64 of a file by memory mapping it and reports progress upon completion.
+    /// </summary>
+    /// <param name="input">The path to the file.</param>
+    /// <param name="job">The job to report progress to.</param>
+    /// <returns>The xxHash64 hash of the file.</returns>
+    public static unsafe Hash XxHash64MemoryMapped(this AbsolutePath input, IActivitySource<Size>? job = null)
+    {
+        #if DEBUG
+        // TODO: Assert that AbsolutePath uses RealFileSystem.
+        // This needs API update in paths, and an corresponding issue.
+        // If it's not RealFileSystem, we can fallback to the stream based version.
+        // Since we can't memory map, unless we add mmap ti Paths library.
```

Comment:
Just add something like `IFileSystem.SupportsMemoryMappedFiles`.

Reply:
Could the `IFileSystem` not technically be used to e.g. redirect the file paths?
I think in this case, checking explicitly for `RealFileSystem` (raw operations) may be preferred.

Reply:
The issue is the code here calls `mmap` directly, avoiding the `IFileSystem `abstraction.

Thus if `IFileSystem` that uses real filesystem would perform remapping, we would get the wrong file.

Which is why I suggested we either make an mmap abstraction in Paths, or simply a way of knowing if I'm working on a path on real FileSystem.

Reply:
> Thus if IFileSystem that uses real filesystem would perform remapping, we would get the wrong file.

How woudl we get the wrong file?